app/src/main/java/ch/epfl/sweng/SDP/home/LeaderboardActivity.java

```diff
@@ -0,0 +1,310 @@
+            friendsButton.setLayoutParams(friendsParams);
+            friendsButton.setScaleType(ImageView.ScaleType.FIT_CENTER);
+
+            // give Button unique Tag to test them later
+            friendsButton.setTag("friendsButton" + index);
+
+            TextView usernameView = new TextView(context);
+            styleView(usernameView, username, getResources().getColor(R.color.colorDrawYellow),
+                    new LinearLayout.LayoutParams(0, LinearLayout.LayoutParams.WRAP_CONTENT, 4));
+
+            TextView trophiesView = new TextView(context);
+            styleView(trophiesView, trophies.toString(),
+                    getResources().getColor(R.color.colorLightGrey),
+                    new LinearLayout.LayoutParams(0,
+                            LinearLayout.LayoutParams.WRAP_CONTENT, 2));
+            trophiesView.setTextAlignment(RelativeLayout.TEXT_ALIGNMENT_TEXT_END);
+            trophiesView.setPadding(0, 0, 30, 0);
+
+            // initializes the friendsButton on first view of leaderboard
+            isFriendWithCurrentUser(context, initializeFriendsButton(friendsButton));
+
+            // modifies friendsButton on every click
+            friendsButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    isFriendWithCurrentUser(context,
+                            changeFriendsButtonBackgroundOnClick(context, friendsButton));
+                }
+            });
+
+            LinearLayout entry = addViews(new LinearLayout(context),
+                    usernameView, trophiesView, friendsButton);
+
+            // set friendsButton to invisible to yourself
+            if(username.equals(Account.getInstance(context).getUsername())) {
+                friendsButton.setVisibility(View.INVISIBLE);
+            }
+
+            entry.setBackgroundColor(Color.DKGRAY);
+            entry.setPadding(30, 10, 30, 10);
+
+            return entry;
+        }
+
+        private LinearLayout addViews(LinearLayout layout, TextView usernameView,
+                              TextView trophiesView, ImageView addFriends) {
+            layout.addView(usernameView);
+            layout.addView(trophiesView);
+            layout.addView(addFriends);
+
+            return layout;
+        }
+
+        private void styleView(TextView view, String text, int color,
+                                         LinearLayout.LayoutParams layoutParams) {
+            view.setText(text);
+            view.setTextSize(25);
+            view.setTextColor(color);
+            view.setTypeface(typeMuro);
+            view.setLayoutParams(layoutParams);
+        }
+
+        /**
+         * Checks if users are friends and applies listener.
+         * @param context   of app
+         * @param listener  how to handle response
+         */
+        private void isFriendWithCurrentUser(final Context context, ValueEventListener listener) {
```

Comment:
This function could also be moved to a separate **proxy** class

Reply:
The Firebase API is already a proxy.

Reply:
You are right